cogs/qrchecker/core.py

```diff
@@ -1,14 +1,39 @@
+from asyncio import Lock
 from logging import getLogger
+from typing import Optional
 
+from PIL import Image
 from redbot.core import Config
 from redbot.core.bot import Red
 
-from .constants import BASE_GUILD
+from .constants import BASE_GLOBAL, BASE_GUILD, KEY_MAX_IMAGE_PIXELS
 
 
 class Core:
     def __init__(self, bot: Red):
         self.bot = bot
+        self.lock = Lock()
         self.logger = getLogger("red.luicogs.QRChecker")
         self.config = Config.get_conf(self, identifier=5842647, force_registration=True)
+        self.config.register_global(**BASE_GLOBAL)
         self.config.register_guild(**BASE_GUILD)
+        self.initialized: bool = False
+        self.bgTask = self.bot.loop.create_task(self.init())
+
+    def cog_unload(self):
+        self.bgTask.cancel()
+
+    async def init(self):
+        await self.setMaxImagePixels()
+        self.initialized = True
+
+    async def setMaxImagePixels(self, value: Optional[int] = None):
+        """Set PIL/Pillow's max image pixels to prevent `DecompressionBombWarning`.
+        If `value` is `None`, the max image pixels value from config will be used."""
+
+        if value is None:
+            value = await self.config.get_attr(KEY_MAX_IMAGE_PIXELS)()
+        else:
+            await self.config.get_attr(KEY_MAX_IMAGE_PIXELS).set(value)
+        Image.MAX_IMAGE_PIXELS = value // 2
```

Comment:
Is this division because we want Pillow to raise `DecompressionBombError` at `value`? The intent is to have the error at the correct value (i.e., twice the threshold set), but I am not fond of the disconnect between the value inputted and the value set, honestly. Given the docstring is saying `DecompressionBombWarning`, how about dropping the division by 2? The warning is raised at `MAX_IMAGE_PIXELS` and the error is raised at twice `MAX_IMAGE_PIXELS`.

Let me know if you have another way to go about this.

Reply:
Correct, I would rather rename the `DecompressionBombWarning` into `Error` instead since it's slightly misleading that the value that is being set is not the one that will raise the error.

Reply:
OK. We can have comments and docstring around here to specify that this is intended to trigger Pillow's `DecompressionBombWarning` and so the value is divided by 2.

Reply:
Updated the comments